toke.c

```diff
@@ -11541,7 +11541,11 @@ S_scan_inputsymbol(pTHX_ char *start)
     return s;
 }
 
-
+/* In the function below, it's quite likely that the calls to memEQ() will have
+ * 'len' == 1.  So create a new macro that adds a conditional to skip the libc
+ * call. */
+#define memEQ_1(a, b, len) (LIKELY(len == 1) ? (*(a) == *(b))       \
+                                             : memEQ(a, b, len))
```

Comment:
For the common 1-length case, `*(a) != *(b)` is more likely to be true, so it will exit the condition earlier, so perhaps:
```
#define memEQ_1(a, b, len) \
    ((*(a) == *(b)) && (((len) == 1) || memEQ((a)+1, (b)+1, (len)-1)))
```
this obviously requires $len >= 1$ but I think that can only be true in the calls in scan_str().